prow/cmd/peribolos/main.go

```diff
@@ -219,36 +219,46 @@ func dumpOrgConfig(client dumpClient, orgName string) (*org.Config, error) {
 		return nil, fmt.Errorf("failed to list teams: %v", err)
 	}
 
-	out.Teams = make(map[string]org.Team, len(teams))
-	for _, t := range teams {
-		p := org.Privacy(t.Privacy)
-		d := t.Description
-		nt := org.Team{
-			TeamMetadata: org.TeamMetadata{
-				Description: &d,
-				Privacy:     &p,
-			},
-			Maintainers: []string{},
-			Members:     []string{},
-		}
-		if t.Parent != nil && t.Parent.ID != 0 {
-			nt.TeamMetadata.ParentTeamID = &t.Parent.ID
-		}
-		maintainers, err := client.ListTeamMembers(t.ID, github.RoleMaintainer)
-		if err != nil {
-			return nil, fmt.Errorf("failed to list team %d(%s) maintainers: %v", t.ID, t.Name, err)
-		}
-		for _, m := range maintainers {
-			nt.Maintainers = append(nt.Maintainers, m.Login)
-		}
-		teamMembers, err := client.ListTeamMembers(t.ID, github.RoleMember)
-		if err != nil {
-			return nil, fmt.Errorf("failed to list team %d(%s) members: %v", t.ID, t.Name, err)
-		}
-		for _, m := range teamMembers {
-			nt.Members = append(nt.Members, m.Login)
+	out.Teams = make(map[string]org.Team)
```

Comment:
I think the general goal here is:

a) Convert github.Team metadata into org.Team metadata
b) Figure out what children a given parent has
c) Create the tree of children, starting at ancestors.

So then we wind up with something like this:
```
names := map[int]string{} // what's the name of a team?
idMap := map[int]org.Team{} // metadata for a team
children := map[int][]int // what children does it have
var tops []int // what are the top-level teams

// first create all the org.Team items
for _, t := range teams {
  nt := org.Team{
      TeamMetadata: org.TeamMetadata{
         Description: t.Description
      },
      Maintainer: client.ListTeamMembers(t.ID)
  }
  names[t.ID] = t.Name
  idMap[t.ID] = nt
  if t.ParentID == 0 { // top level team
    tops = append(tops, t.ID)
  } else { // add this id to the list of the parent's children
    children[t.ParentID] = append(children[t.ParentID], t.ID)
  }
}

makeChild := func(id int) org.Team {
  t := idMap[id]
  for _, cid := range children[id] {
    child := makeChild(cid)
    t.Children = append(t.Children, child)
  }
  return t
} 

for _, id := range tops {
  out.Teams[names[id]] = makeChild(id)
}
```